clang/lib/CodeGen/MicrosoftCXXABI.cpp

```diff
@@ -1110,6 +1110,10 @@ static bool isTrivialForMSVC(const CXXRecordDecl *RD, QualType Ty,
       isa<VectorType>(Base)) {
     return true;
   }
+  if (CGM.getTarget().getTriple().isX86() &&
+      CGM.getABIInfo().isHomogeneousAggregate(Ty, Base, NumElts)) {
```

Comment:
This looks like it might affect non-vectorcall calls; can you check (and add a testcase)?

Please also add test coverage for 32-bit x86.